AmplifyPlugins/DataStore/Tests/DataStoreHostApp/AWSDataStorePluginLazyLoadTests/LL1/AWSDataStoreLazyLoadPostComment4V2StressTests.swift

```diff
@@ -0,0 +1,211 @@
+            }
+            for i in 0..<count {
+                let post = Post(title: "title")
+                let comment = Comment(content: "content", post: post)
+                savedModels.append((comment,post))
+
+                Task.detached {
+                    Amplify.Logging.info("[\(AWSDataStoreLazyLoadPostComment4V2Tests.loggingContext)] Saving comment and post, index: \(i)")
+                    do {
+                        _ = try await Amplify.DataStore.save(post)
+                        _ = try await Amplify.DataStore.save(comment)
+                    } catch {
+                        // This is expected to happen when DataStore is interrupted and did not save the MutationEvent.
+                        Amplify.Logging.info("[\(AWSDataStoreLazyLoadPostComment4V2Tests.loggingContext)] Failed to save post and/or comment, post: \(post.id). comment: \(comment.id). error \(error)")
+                    }
+                }
+                Task.detached {
+                    Amplify.Logging.info("[\(AWSDataStoreLazyLoadPostComment4V2Tests.loggingContext)] Stop/Start, index: \(i)")
+                    try await Amplify.DataStore.stop()
+                    try await Amplify.DataStore.start()
+                }
+                self.index = i
+                try await Task.sleep(seconds: 0.01)
+            }
+        }
+
+        /// Subscribe to DataStore Hub events, and handle `OutboxStatusEvent`'s.
+        /// Maintain the latest state of whether the outbox is empty or not in `isOutboxEmpty` variable.
+        /// Fulfill `savesSyncedExpectation` after all tasks have been created and the outbox is empty.
+        private func subscribeToOutboxEvent(_ savesSyncedExpectation: XCTestExpectation) {
+            self.subscribeToOutboxEventTask = Task {
+                for await event in Amplify.Hub.publisher(for: .dataStore).values {
+                    switch event.eventName {
+                    case HubPayload.EventName.DataStore.outboxStatus:
+                        guard let outboxEvent = event.data as? OutboxStatusEvent else {
+                            return
+                        }
+                        isOutboxEmpty = outboxEvent.isEmpty
+                        outboxEventsCount += 1
+                        Amplify.Logging.info("[\(AWSDataStoreLazyLoadPostComment4V2Tests.loggingContext)] \(outboxEventsCount) isOutboxEmpty: \(isOutboxEmpty), index: \(index)")
+                        if index == (count - 1)  && isOutboxEmpty {
+                            XCTAssertEqual(savedModels.count, count)
+                            savesSyncedExpectation.fulfill()
+                        }
+                    default:
+                        break
+                    }
+                }
+            }
+        }
+    }
+
+    func assertQueryComment(_ savedComment: Comment, post: Post) async throws {
+        guard let persistedComment = try await Amplify.DataStore.query(
```

Comment:
i think we can add a check here that if `persistedComment` is not in DB, then it should not have synced as well through `API.query()` call

Reply:
updated